File: preprocess/mesh.py
import numpy as np

class Mesh:
    """ 1-D mesh creation """

    def __init__(self, n_el, data):
        self.vols_ID = np.arange(n_el)
        self.vols_coord = np.linspace(0+1/n_el,1-data['mesh_length']/n_el,n_el)
        self.internal_faces_ID = np.arange(n_el-1)
        self.vols_contours_ID = np.array([0,n_el-1])
        self.neighboring_vols(n_el)

    def neighboring_vols(self, n_el):
        self.conec = np.empty((n_el-1,2),dtype=int)
        self.conec[:,0] = np.arange(n_el-1)
        self.conec[:,1] = np.arange(1,n_el)

File: Mesh.m
% Arquivo criado para gerar a malha (gerada manualmente) - só funciona para
% malhas 1-D.
classdef Mesh
    properties
        n_el 
        L 
    end
    methods 
        function conec = faces_conec(obj)
            conec = zeros(obj.n_el-1,2);
            conec(:,1) = linspace(1,obj.n_el-1, obj.n_el-1);
            conec(:,2) = linspace(2,obj.n_el, obj.n_el-1);
        end
        function vols_ID = all_vols_ID(obj)
            vols_ID = linspace(1,obj.n_el, obj.n_el);
        end
        function contour_vols_ID = contour_vols_ID(obj)
            contour_vols_ID = [1 obj.n_el];
        end
        function vols_coord = vols_coord(obj)
            vols_coord = linspace(0+obj.L/(2*obj.n_el), 1 - obj.L/(2*obj.n_el),obj.n_el); 
        end
        function faces_ID = internal_faces_ID(obj)
            faces_ID = linspace(1,obj.n_el);
        end
    end
end